angular-app/src/app/services/UserService.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { catchError, Observable, tap, throwError } from 'rxjs';
+
+@Injectable()
+export class UserService {
+
+    private readonly apiUrl = 'http://localhost:8080/api';
```

Comment:
I'm pretty sure you can make URLs relative to root using either `~/api` or `/api`, which would make this more robust, as we won't have to change it once we move into production